src/lib.rs

```diff
@@ -292,6 +312,30 @@ impl From<JsonFrontendError> for Error {
 ///
 /// [`BpfProgram`]: type.BpfProgram.html
 pub fn apply_filter(bpf_filter: BpfProgramRef) -> Result<()> {
+    apply_filter_with_flags(bpf_filter, 0)
+}
+
+/// Apply a BPF filter to the all threads in the process via the TSYNC feature. Please read the
+/// man page for seccomp (`man 2 seccomp`) for more information.
+///
+/// # Arguments
+///
+/// * `bpf_filter` - A reference to the [`BpfProgram`] to be installed.
+///
+/// [`BpfProgram`]: type.BpfProgram.html
+pub fn apply_filter_all_threads(bpf_filter: BpfProgramRef) -> Result<()> {
+    apply_filter_with_flags(bpf_filter, libc::SECCOMP_FILTER_FLAG_TSYNC)
+}
+
+/// Apply a BPF filter to the calling thread.
+///
+/// # Arguments
+///
+/// * `bpf_filter` - A reference to the [`BpfProgram`] to be installed.
+/// * `flags` - A u64 representing a bitset of seccomp's flags parameter.
+///
+/// [`BpfProgram`]: type.BpfProgram.html
+fn apply_filter_with_flags(bpf_filter: BpfProgramRef, flags: libc::c_ulong) -> Result<()> {
```

Comment:
Ideally `flags` should be similar to something like OFlags in nix: https://docs.rs/nix/latest/nix/fcntl/struct.OFlag.html, it's slightly more robust, but if `apply_filter_with_flags` remain just for internal purposes (ie. `flags` is not part of the public API), then leaving it like this for now is fine.

Reply:
This was the initial approach in the PR but since different flags for the seccomp syscall results in different return values and has different implications, I suggested to keep this function for internal use only, to keep things simple and hard to misuse